homeassistant/helpers/service.py

```diff
@@ -189,7 +190,22 @@ def async_prepare_call_from_config(
 
     domain, service = domain_service.split(".", 1)
 
-    target = config.get(CONF_TARGET)
+    target = {}
+    if CONF_TARGET in config:
+        conf = config.get(CONF_TARGET)
+        try:
+            template.attach(hass, conf)
+            target.update(template.render_complex(conf, variables))
+            if CONF_ENTITY_ID in target:
+                cv.comp_entity_ids(target[CONF_ENTITY_ID])
```

Comment:
We should also store the output as it converts strings to a list. 
```suggestion
                target[CONF_ENTITY_ID] = cv.comp_entity_ids(target[CONF_ENTITY_ID])
```

Reply:
Oh right!

Reply:
Oh wait, is validation even necessary? It will be validated when we call the service?